testing/dart_isolate_runner.h

```diff
@@ -0,0 +1,198 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/make_copyable.h"
+#include "flutter/fml/paths.h"
+#include "flutter/fml/synchronization/waitable_event.h"
+#include "flutter/fml/thread.h"
+
+namespace flutter {
+namespace testing {
+
+class AutoIsolateShutdown {
+ public:
+  AutoIsolateShutdown() = default;
+
+  AutoIsolateShutdown(std::shared_ptr<DartIsolate> isolate,
```

Comment:
Now that this is a general purpose utility, please OOL these like in the rest of the engine codebase.

Reply:
Fixed these in #16941